### PR TITLE
Adding HandleOnlyMyJobsPermission as generic permission to execute me…

### PR DESCRIPTION
**Pull request by marcocast**
…thods of the SchedulerFrontend on other user's jobs

### CODE REVIEW

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java [hunk 1]
```diff
@@ -1114,9 +1110,7 @@ public String getTaskServerLogsByTag(String jobId, String taskTag)
     @Override
     @ImmediateService
     public List<SchedulerUserInfo> getUsers() throws NotConnectedException, PermissionException {
-        UserIdentificationImpl ident = frontendState.checkPermission("getUsers",
-                "You don't have permissions to get users");
-        frontendState.checkOwnStatePermission(false, ident);
+        frontendState.checkPermission("getUsers", "You don't have permissions to get users");
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java]: how are modified the getUsers method calls ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java [hunk 2]
```diff
@@ -288,13 +288,9 @@ synchronized SchedulerState getState(boolean myJobsOnly)
...
         ListeningUser ui = identifications
                 .get(PAActiveObject.getContext().getCurrentRequest().getSourceBodyID());
-        try {
-            checkOwnStatePermission(myJobsOnly, ui.getUser());
-            return myJobsOnly ? sState.filterOnUser(ui.getUser().getUsername()) : sState;
-        } catch (PermissionException ex) {
-            logger.info(ex.getMessage());
-            throw ex;
-        }
+
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: I'm afraid, that by this modification, a user will be able to see all other users jobs (task list)

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java [hunk 3]
```diff
@@ -547,6 +538,25 @@ synchronized void checkChangeJobPriority(JobId jobId, JobPriority priority) thro
         }
     }
 
+    synchronized void checkPermissions(String methodName, IdentifiedJob identifiedJob, String errorMessage)
+            throws NotConnectedException, UnknownJobException, PermissionException {
+        try {
+            checkJobOwner(methodName, identifiedJob, errorMessage);
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: as this method is now called only at one place, I think it's not necessary any more to handle it with exceptions.

Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: This method should throw an exception, like it was before. so all the existing code will not be changed.

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java (hunk 1 above)
Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontend.java]: what you mean?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java (hunk 2 above)
Comment (marcocast) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java]: you mean this particular method or this change in general?

**Comment by marcocast** [comment]
@fviale ok, i ll externalise the error messages in a class

**Pull request merged**